src/components/core/a11y/focus.ts

```diff
@@ -111,30 +111,26 @@ export class FocusHandler {
     );
   }
 
-  public setDataHasVisibleFocusWithinListener(element: HTMLElement): void {
-    // Determine whether the element has a visible focus within.
-    getFocusableElements(Array.from(element.children) as HTMLElement[], {
-      includeInvisibleElements: true,
-    })?.forEach((el) => {
-      el.addEventListener(
-        'focusin',
-        () => {
-          toggleDatasetEntry(
-            element,
-            'hasVisibleFocusWithin',
-            sbbInputModalityDetector.mostRecentModality === 'keyboard',
-          );
-        },
-        { signal: this._controller.signal },
-      );
-      el.addEventListener(
-        'blur',
-        () => {
-          toggleDatasetEntry(element, 'hasVisibleFocusWithin', false);
-        },
-        { signal: this._controller.signal },
-      );
-    });
+  // Determine whether the element has a visible focus within.
```

Comment:
Nice! I think we could now just move it into an own file an create a ReactiveController, so we could just use it in the components where needed, without the need to connect or disconnect it manually. What do you think?
I feel it doesn't belong to the focushandler as it doesn't use any field of it. BTW: FocusHandler could be a ReactiveController as well :)